compiler/rustc_hir_analysis/src/astconv/mod.rs

```diff
@@ -1440,6 +1441,14 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                             tcx.associated_items(pred.def_id())
                                 .in_definition_order()
                                 .filter(|item| item.kind == ty::AssocKind::Type)
+                                .filter(|item| {
```

Comment:
What is this for?

Reply:
Oh -- we should definitely be bailing out in the case that our trait has RPIT associated items and we're trying to make an object out of it. This is probably not the right approach?

Regardless, any modification like this needs some sort of comment justifying it.

Reply:
Removed, it will be easier to justify with tests later.